CONTRIBUTING.md

```diff
@@ -0,0 +1,110 @@
+1. **Fork the repository** and create a new branch from `main`.
+2. **Follow these instructions** to install Kantra: [Setting up Kantra and developing rules](https://github.com/konveyor/kantra/blob/main/docs/rules-quickstart.md#how-to-install-kantra).
+3. Make your changes according to the [Rule creation guidelines](#4-rule-creation-guidelines), test them, and commit your work.
+
+### Rule creation guidelines
+There are a few conventions that we follow when creating rules:
+
+#### Directory structure
+Rules can be found within the `default/generated` folder. Within this folder, rules are organized by target technology.
+**_Source_** and **_target_** technologies are the technologies to migrate **_to_** and **_from_**, respectively.
+If a new target technology is to be added, a new folder should be created to cover for it. Within each target directories,
+additional directories can be created to further segregate the rules according to their source technologies. For instance,
+Within the `eap8` target folder, we could have `eap6` and `eap7` folders covering for migration to eap8 from eap 6 and 7
+respectively.
+
+Additionally, within each rules directory, a `test` directory should be created to place the rules's tests. Since each
+test needs data, a `data` directory must also be created inside `test`. For instance:
+
+```shell
+.
+└── target
+    ├── source1
+    │   ├── rules-1.yaml
+    │   ├── rules-2.yaml
+    │   └── tests
+    │       ├── data
+    │       │   ├── rules-1
+    │       │   └── rules-2
+    │       ├── rules-1.test.yaml
+    │       └── rules-2.test.yaml
+    └── source2
+        └── ...
+```
+Inside of each `data` folder, it is recommended to segregate the test data, so that each folder within it contains the
+data for the tests of a single rules file. This way, other tests will be less affected by changes in the data.
+
+#### Tests organization and file naming
+- It is recommended that a single rules file holds rules for the migration of a related set of concerns within its particular
+migration path. For instance, if migration from `A` to `B` implies changes related to data sources and web services,
+two different rule files should be created, one containing the rules related to data sources and another one for web services. 
+This makes it easier to find rules, create tests and data for them, and keep the structure organized in general.
+
+- Rules should be named using hypens for word separation. Also, names for the rule files should indicate the target
+technology and the concern of the rules they contain. If needed, they can also indicate the source technology.
+
+- Test files should be named after the rule file they cover in the following way: `<rule-file>.test.yaml`
+
+#### Rule IDs
+Rule IDs must be unique within the rulesets being used, and contain in themselves the name of the rule file plus a
+number. These numbers must have 5 digits and increase by 10 each time, to follow a convention set in the old Windup rules.
```

Comment:
Why do we have to follow this convention set? It's confusing. Can't we just enforce this to be an incrementing integer?

Reply:
We can, although then we would have two different conventions. It's not super important, but just wanted to be homogeneous.